phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/DelegateRegionScanner.java

```diff
@@ -97,4 +152,10 @@ public RegionScanner getNewRegionScanner(Scan scan) throws IOException {
             throw new DoNotRetryIOException(e);
         }
     }
+
+    public static class ScannerContextNextNotImplementedException extends IOException {
```

Comment:
Why is this necessary?

Reply:
It can be used to inform DelegateRegionScanner that a lower level does not implement next/nextRaw with scannerContext, for example in this change in HashJoinRegionScanner: https://github.com/apache/phoenix/pull/1936/files#diff-77ea25574d36a1657d82cedf3384ebca3d8b11ac7a0de74dd5bcb18c59069afcR352

Reply:
Why does HashJoinRegionScanner not support it? Should all layers support this now without exception?

Reply:
`HashJoinRegionScanner.nextRaw` supports it. Should `next()` also support it?

Reply:
Yes, all should support it from now on

Reply:
What should be the implementation for next in HashJoinRegionScanner, with and without ScannerContext? Should it just call the corresponding nextRaw?

Reply:
I made the change in the latest commit.